tests/examples/CMakeLists.txt

```diff
@@ -0,0 +1,251 @@
+function(escape_string variable content escaped_content line_prefix)
+    # If DOC_HIDE_ALL is present, do nothing
+    if(variable MATCHES "--DOC_HIDE_ALL")
+        return()
+    endif()
+
+    string(REGEX REPLACE "\n" ";" var_lines "${variable}")
+
+    set(tmp_output ${content})
+    foreach (LINE ${var_lines})
+        if(NOT LINE MATCHES "^.+--DOC_[A-Z]+$")
+            set(tmp_output ${tmp_output}\n--${line_prefix}${LINE})
+        endif()
+    endforeach()
+
+    set(${escaped_content} ${tmp_output} PARENT_SCOPE)
+endfunction()
+
+# Extract lines with the --DOC_HEADER marker
+function(extract_header variable pre_output post_output)
+    string(REGEX REPLACE "\n" ";" var_lines "${variable}")
+
+    set(IS_POST 0)
+
+    foreach (LINE ${var_lines})
+        # This function doesn't escape the lines, so make sure they are
+        # already comments.
+        if(LINE MATCHES "^--.*--DOC_HEADER")
+            # Remove the header tag
+            string(REGEX REPLACE "[ ]*--DOC_HEADER" "" LINE "${LINE}")
+
+            # ldoc is picky about what happen after the first --@, so split
+            # the output between all that come before and all that come after.
+            if (NOT IS_POST AND LINE MATCHES "^--[ ]*@")
+                set(IS_POST 1)
+            endif()
+
+            if (IS_POST)
+                set(tmp_post_output ${tmp_post_output}\n${line_prefix}${LINE})
+            else()
+                set(tmp_pre_output ${tmp_pre_output}\n${line_prefix}${LINE})
+            endif()
+        endif()
+    endforeach()
+
+    set(${post_output} "${tmp_post_output}\n--" PARENT_SCOPE)
+    set(${pre_output} "${tmp_pre_output}\n--" PARENT_SCOPE)
+endfunction()
+
+# Read a code file and convert it to ldoc usage example.
+#  * add "--" in front of each lines
+#  * drop empty lines
+#  * convert " " lines into empty lines
+#  * drop lines ending with "--DOC_HIDE"
+function(escape_code path escaped_content pre_header post_header)
+    file(READ ${path} path)
+
+    escape_string("${path}\n" "" escaped_code "")
+
+    extract_header("${path}" example_pre_header example_post_header)
+
+    set(${escaped_content} ${escaped_code} PARENT_SCOPE)
+    set(${pre_header} ${example_pre_header} PARENT_SCOPE)
+    set(${post_header} ${example_post_header} PARENT_SCOPE)
+endfunction()
+
+# Execute a lua file.
+function(run_test test_path namespace template escaped_content)
+
+    # A template is required to know how to handle the output.
+    if (template STREQUAL " ")
+        message(FATAL_ERROR "No template found for " ${test_path} ", bye")
+    endif()
+
+    # Get the file name without the extension
+    get_filename_component(${test_path} TEST_FILE_NAME NAME)
+    set(IMAGE_PATH "${IMAGE_DIR}/${namespace}_${TEST_FILE_NAME}")
+
+    # Execute the script, leave the image extension decision to the test
+    # SVG is preferred, but PNG is better suited for some tests, like bitmap
+    # patterns.
+    execute_process(
```

Comment:
Should there be a `message(STATUS "Doing foo")` here, so that we know that it is executing something like this? If not, should there be such a message at the beginning and at the end of the file so that one can at least see how much time is spent here in the build?